File: js/collisionDetector.js
detection(physicObject1, physicObject2) {
    dabs12 = abs(physicObject1.getAbs() - physicObject2.getAbs())
    dord12 = abs(physicObject1.getOrd() - physicObject2.getOrd())

    minAbsDist = physicObject1.getHalfWidth() - physicObject2.getHalfWidth()
    minOrdDist = physicObject1.getHalfHeight() - physicObject2.getHalfHeight()

    if (dabs12 < minAbsDist || dord12 < minOrdDist ) {
        return true
    }

    return false

}

File: py/collisionDetector.py
import physicObject

def rectangleCollisionDetector(pObject1, pObject2):
    absDist = abs(pObject1.getAbs() - pObject2.getOrd)
    ordDist = abs(pObject1.getOrd() - pObject2.getOrd())

    minAbsDist = pObject1.getHalfWidth() + pObject2.getHalfWidth
    minOrdDist = pObject1.getHalfHeight() + pObject2.getHalfHeight

    return absDist >= minAbsDist and ordDist >= minOrdDist


